public/js/controllers/game.js

```diff
@@ -289,4 +291,36 @@ angular.module('mean.system')
       })
     }
   };
+  $scope.$watchCollection('chat.messageArray', (newValue, oldValue) => {
+      $timeout(() => {
+        $scope.scrollChatThread();
+      }, 100);
+    });
+    $scope.scrollChatThread = () => {
+      const chatResults = document.getElementById('chatbox');
+      chatResults.scrollTop = chatResults.scrollHeight;
+    };
+   /**
+    * Method to send messages
+    * @param{String} userMessage - String containing the message to be sent
+    * @return
+    */
+    $scope.sendMessage = (userMessage) => {
+      $scope.chat.postGroupMessage(userMessage);
+      $scope.chatMessage = '';
+    };
+    $scope.keyPressed = function ($event) {
+      const keyCode = $event.which || $event.keyCode;
+      if (keyCode === 13) {
+        $scope.sendMessage($scope.chatMessage);
+      }
+    };
+
+    $scope.showChat = function () {
```

Comment:
Do you want to rename this method to `toggleChatWindow` ? 

I believe that is more expressive.